app/src/main/java/org/oppia/app/topic/TopicActivity.kt

```diff
@@ -48,18 +49,23 @@ class TopicActivity : InjectableAppCompatActivity(), RouteToQuestionPlayerListen
     startActivity(RevisionCardActivity.createRevisionCardActivityIntent(this, topicId, subtopicId))
   }
 
-  override fun routeToExploration(internalProfileId: Int, topicId: String, storyId: String, explorationId: String) {
+  override fun routeToExploration(internalProfileId: Int, topicId: String, storyId: String, explorationId: String, backflowScreen: Int?) {
     startActivity(
       ExplorationActivity.createExplorationActivityIntent(
         this,
         internalProfileId,
         topicId,
         storyId,
-        explorationId
+        explorationId,
+        backflowScreen
       )
     )
   }
 
+  override fun onBackPressed() {
+    startActivity(HomeActivity.createHomeActivity(this,internalProfileId))
```

Comment:
Spacing issue: `startActivity(HomeActivity.createHomeActivity(this, internalProfileId))`

Reply:
Done